HTML cleanup and subexpression lookup

javdejong opened the pull request:
I liked newlines for different subexpressions, otherwise users might have a hard time figuring out what pronunciation goes with what expression. I did it like this for the cards:
![image](https://user-images.githubusercontent.com/3198903/49341214-9acdee00-f64a-11e8-9ca2-63f4aea77b93.png)

And like this for the manual lookup:
![image](https://user-images.githubusercontent.com/3198903/49341200-77a33e80-f64a-11e8-9573-f24a9f079a06.png)

Of course, if people like different separators, we can always add that as a customization option to the config file later. For now I think sane defaults should be good enough.

javdejong commented:
@weirdalsuperfan If you could check the commits. I mostly just shuffled your code around, or simplified it where possible. Going to the "Changes" tab might be a bit much, but you can check each commit individually in the "Commits" tab.

SpongebobSquamirez commented:
Yeah, I'll check it ASAP. Might not be able to before the weekend, but I definitely will by the end of the weekend.

SpongebobSquamirez commented:
@javdejong I took a look at the PR and I have two changes to make:
- you should specify the encoding of the config file to io.open as utf-8 (line 45); I got an error otherwise
- bulk pronunciation should be using getFormattedPronunciations and not merging with *** afterwards, just writing the result directly to the field

But I'm going to hold off on pushing those until I get colorization working. I finished all my other tests on this pull request, so I think it is working pretty well, but that's all I really had time to do this weekend.

I also don't think manual lookup works properly. It might have something to do with Windows/cp65001 encoding (it was working before):
![nhk_manual_lookup](https://user-images.githubusercontent.com/11866755/49695371-6673a800-fbdd-11e8-8f77-369b9b9742ad.PNG)

I can't see where the difference might be coming from, but it seems like it's not related to the japanese support addon, but more likely to either line 417 or 435.

javdejong commented:
Thanks for checking! I fixed the utf8 encoding on master (as it's a bug unrelated to this PR and the fix is indisputable). I also fixed the bulk pronunciations as a "fixup" commit now, which I will squash before merging to master. Same for the manual lookup thing you mentioned (which was due to a missing `u` to make a string unicode).

javdejong commented:
I'll go ahead and merge this.
